File: tests/example-frames/elements/light-intensity.c
// $dep ufbx light.fbx
// $ clang ufbx.c main.c -lm -o example
// $ ./example
#include <stdio.h>
#include <assert.h>
#include "ufbx.h"

// -- EXAMPLE_SOURCE --

int main(int argc, char **argv)
{
    ufbx_scene *scene = ufbx_load_file("light.fbx", NULL, NULL);
    assert(scene);

    ufbx_node *node = ufbx_find_node(scene, "Light");
    assert(node);
    assert(node->light);

    print_intensity(node->light);

    return 0;
}

File: tests/example-frames/elements/light-intensity.rs
// $dep light.fbx
// $ cargo build
// $ cargo run --quiet
use ufbx;

// -- EXAMPLE_SOURCE --

fn main() {
    let scene = ufbx::load_file("light.fbx", Default::default()).expect("failed to load scene");
    let node = scene.find_node("Light").expect("could not find node");
    let light = node.light.as_ref().expect("node did not contain a light");
    print_intensity(light);
}
